cupy/cusparse.py

```diff
@@ -117,6 +117,60 @@ def cscsort(x):
         P.data.ptr, cusparse.CUSPARSE_INDEX_BASE_ZERO)
 
 
+def coosort(x):
+    handle = device.get_cusparse_handle()
+    m, n = x.shape
+    nnz = x.nnz
+
+    buffer_size = cusparse.xcoosort_bufferSizeExt(
+        handle, m, n, nnz, x.row.data.ptr, x.col.data.ptr)
+    buf = cupy.empty(buffer_size, 'b')
+    P = cupy.empty(nnz, 'i')
+    cusparse.createIdentityPermutation(handle, nnz, P.data.ptr)
+    cusparse.xcoosortByRow(
+        handle, m, n, nnz, x.row.data.ptr, x.col.data.ptr,
+        P.data.ptr, buf.data.ptr)
+    _call_cusparse(
+        'gthr', x.dtype,
+        handle, nnz, x.data.data.ptr, x.data.data.ptr,
+        P.data.ptr, cusparse.CUSPARSE_INDEX_BASE_ZERO)
+
+
+def coo2csr(x):
+    handle = device.get_cusparse_handle()
+    m = x.shape[0]
+    indptr = cupy.empty(m + 1, 'i')
+    cusparse.xcoo2csr(
+        handle, x.row.data.ptr, x.nnz, m,
+        indptr.data.ptr, cusparse.CUSPARSE_INDEX_BASE_ZERO)
+    return cupy.sparse.csr.csr_matrix(
+        (x.data, x.col, indptr), shape=x.shape)
+
+
+def csr2coo(x, data, indices):
+    """Convert a CSR-matrix to COO format.
+
+    Args:
+        x (cupy.sparse.csr_matrix): A matrix to be converted.
+        data (cupy.ndarray): A data array for converted data.
+        indices (cupy.ndarray): A index array for converted data.
+
+    Returns:
+        cupy.sparse.coo_matrix: Returns a converted matrix.
+
+    """
+    handle = device.get_cusparse_handle()
+    m = x.shape[0]
+    nnz = len(x.data)
+    row = cupy.empty(nnz, 'i')
```

Comment:
It must be int32: http://docs.nvidia.com/cuda/cusparse/index.html#cusparse-lt-t-gt-csr2coo